src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,719 @@
+    auto heteroNode = doc.append_child("hetero");
+    heteroNode.append_attribute("name").set_value(m_name.c_str());
+
+    auto inputs_map_node = heteroNode.append_child("inputs_to_submodels_inputs");
+    for (const auto& it : m_inputs_to_submodels_inputs) {
+        auto xml_node = inputs_map_node.append_child("pair");
+        xml_node.append_attribute("submodel_idx").set_value(std::to_string(it.first).c_str());
+        xml_node.append_attribute("node_idx").set_value(std::to_string(it.second).c_str());
+    }
+    auto outputs_map_node = heteroNode.append_child("outputs_to_submodels_outputs");
+    for (const auto& it : m_outputs_to_submodels_outputs) {
+        auto xml_node = outputs_map_node.append_child("pair");
+        xml_node.append_attribute("submodel_idx").set_value(std::to_string(it.first).c_str());
+        xml_node.append_attribute("node_idx").set_value(std::to_string(it.second).c_str());
+    }
+
+    auto submodels_input_to_prev_output_node = heteroNode.append_child("submodels_input_to_prev_output");
+    for (const auto& it : m_submodels_input_to_prev_output) {
+        auto xml_node = submodels_input_to_prev_output_node.append_child("record");
+        xml_node.append_attribute("in_submodel_idx").set_value(std::to_string(it.first.first).c_str());
+        xml_node.append_attribute("in_node_idx").set_value(std::to_string(it.first.second).c_str());
+        xml_node.append_attribute("out_submodel_idx").set_value(std::to_string(it.second.first).c_str());
+        xml_node.append_attribute("out_node_idx").set_value(std::to_string(it.second.second).c_str());
+    }
+
+    auto subnetworksNode = heteroNode.append_child("compiled_submodels");
+    for (const auto& comp_model_desc : m_compiled_submodels) {
+        auto sub_comp_model = comp_model_desc.compiled_model;
+        OPENVINO_ASSERT(sub_comp_model);
+
+        auto subnetworkNode = subnetworksNode.append_child("compiled_submodel");
+        subnetworkNode.append_attribute("device").set_value(comp_model_desc.device.c_str());
+    }
+
+    auto heteroConfigsNode = heteroNode.append_child("hetero_config");
+    for (const auto& config : m_cfg.get_hetero_properties()) {
+        auto heteroConfigNode = heteroConfigsNode.append_child("config");
+        heteroConfigNode.append_attribute("key").set_value(config.first.c_str());
+        heteroConfigNode.append_attribute("value").set_value(config.second.as<std::string>().c_str());
+    }
+
+    doc.save(model_stream, nullptr, pugi::format_raw);
+    doc.reset();
+    model_stream << std::endl;
+
+    for (const auto& comp_model_desc : m_compiled_submodels) {
+        if (get_plugin()->get_core()->device_supports_model_caching(comp_model_desc.device)) {
+            comp_model_desc.compiled_model->export_model(model_stream);
+        } else {
+            auto model = comp_model_desc.model;
+            if (!model)
+                OPENVINO_THROW("Hetero device supports only OpenVINO Model representation");
```

Comment:
do we need it? what other representation is expected here?

Reply:
Decided to leave exception, message is updated